main/src/main/java/org/mobilitydata/gtfsvalidator/util/CalendarUtil.java

```diff
@@ -0,0 +1,159 @@
+package org.mobilitydata.gtfsvalidator.util;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Multimaps;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendar;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarDate;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarDateExceptionType;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarDateTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarTableContainer;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedSet;
+
+/**
+ * Provides a collection of functions to work with service dates in <i>calendar.txt</i> and <i>calendar_dates.txt</i>.
+ */
+public final class CalendarUtil {
+    private CalendarUtil() {
+    }
+
+    /**
+     * Extracts the service period information of a service from GTFS schema.
+     *
+     * @param calendar      the row in <i>calendar.txt</i> or null
+     * @param calendarDates the list of rows in <i>calendar_dates.txt</i>, may
+     *                      be empty
+     * @return a {@link ServicePeriod} object
+     */
+    public static ServicePeriod
```

Comment:
Formatting looks off here - method name should be on same line

Reply:
This formatting was applied automatically by clang-format and IDEA. Let the machines format the code, we have more interesting stuff to do ;)